### ISSUE TITLE
Service with invalid host_name causes 0 services to load.

### ISSUE THREAD

**Issue by dlgood23** [naemon/naemon-core#169]
We have a naemon 1.0.5 instance where an admin accidentally used 'host_name' instead of 'hostgroup_name' in a single service definition.  Naemon generated a couple of ERROR messages but did not include the error in the final tally and did not abort, ending up loading a configuration with 0 services (instead of more than 14,000) resulting in us losing all retention data (acknowledgements, comments, disabled notifications) for those services.  Here's the output of 'naemon -v' (we've since deleted the deprecated items from our config):

```
# naemon -v naemon.cfg

Naemon Core 1.0.5-pkg
Copyright (c) 2013-present Naemon Core Development Team and Community Contributors
Copyright (c) 2009-2013 Nagios Core Development Team and Community Contributors
Copyright (c) 1999-2009 Ethan Galstad
License: GPL

Website: http://www.naemon.org
Reading configuration data...
Warning: enable_environment_macros is deprecated and will be removed.
Warning: use_large_installation_tweaks is deprecated and will be removed. Naemon should always be fast
Warning: daemon_dumps_core is deprecated and will be removed. Use system facilities to control coredump behaviour instead
Warning: max_check_result_file_age is deprecated and will be removed. Support for processing check results from disk will be removed
Warning: check_result_path is deprecated and will be removed. Support for processing check results from disk will be removed
Warning: max_check_result_reaper_time is deprecated and will be removed. Support for processing check results from disk will be removed
Warning: check_result_reaper_frequency is deprecated and will be removed. Support for processing check results from disk will be removed
Warning: naemon_group is deprecated and will be removed. Naemon is compiled to be run as naemon:naemon
Warning: naemon_user is deprecated and will be removed. Naemon is compiled to be run as naemon:naemon
   Read main config file okay...
Error: Could not find any host matching 'srv-linux' (config file '/etc/naemon/conf.d/linux.cfg', starting on line 68)
Error: Failed to expand host list 'srv-linux' for service 'CPU INFO' (/etc/naemon/conf.d/linux.cfg:68)
   Read object config files okay...

Running pre-flight check on configuration data...

Checking objects...
        Checked 0 services.
        Checked 511 hosts.
        Checked 5 contacts.
        Checked 151 host groups.
        Checked 4 service groups.
        Checked 3 contact groups.
        Checked 163 commands.
        Checked 6 time periods.
        Checked 0 host escalations.
        Checked 0 service escalations.
Checking for circular paths...
        Checked 511 hosts
        Checked 0 service dependencies
        Checked 0 host dependencies
        Checked 6 timeperiods
Checking global event handlers...
Checking obsessive compulsive processor commands...
Checking misc settings...

Total Warnings: 0
Total Errors:   0

Things look okay - No serious problems were detected during the pre-flight check

```

**Comment by sni** [comment]
This sounds familiar, i'd say this is already fixed in the 1.0.6. Have you tried the latest release?

**Comment by dlgood23** [comment]
We just upgraded to 1.0.5 about a month ago, so we probably won't be able to schedule that.  I might be able to spin up a VM to test it on, though.

**Comment by dlgood23** [comment]
I've verified that this is fixed in 1.0.6.  Thanks!
